File: src/main/java/com/happn/worshop/infrastructure/UserInMemoryRepository.java
package com.happn.worshop.infrastructure;

import com.happn.worshop.domain.User;
import com.happn.worshop.domain.UserRepository;
import org.springframework.stereotype.Repository;

import java.util.HashMap;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;

@Repository
public class UserInMemoryRepository implements UserRepository {

    private Map<UUID, User> users = new HashMap<>();

    @Override
    public Optional<User> findById(UUID id) {
        return Optional.ofNullable(users.get(id));
    }

    @Override
    public User save(User user) {
        users.put(user.id(), user);
        return user;
    }
}

File: 2025-05-06_Spring-101-Kotlin-Gradle/src/main/kotlin/com/happn/workshop/infra/UserInMemoryRepository.kt
package com.happn.workshop.infra

import com.happn.workshop.domain.User
import com.happn.workshop.domain.UserCreate
import com.happn.workshop.domain.UserRepository
import com.happn.workshop.domain.toUser
import org.springframework.stereotype.Repository
import java.util.*

@Repository
class UserInMemoryRepository : UserRepository {

    private val users = mutableMapOf<UUID, User>()

    override fun findById(id: UUID): User? = users[id]

    override fun save(user: UserCreate): String = UUID.randomUUID().apply {
        users.put(this, user.toUser(id = this))
    }.toString()

}